app/models/cart_chassis.rb

```diff
@@ -0,0 +1,307 @@
+# frozen_string_literal: true
+#
+# Copyright 2021 Victoria Garcia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A CartChassis is a Plain Old Ruby Object (with no presence of its own
+# in the database) that consists of two carts (each of which DOES have a
+# a presence in the database).  One cart, with the status for_now, contains
+# the user's current items.  The other cart, with the status for_later,
+# contains items the user isn't ready to purchase yet.
+
+# When the user is ready to pay (or to reserve, pending payment by cheque)
+# the user's for_now cart goes into processing and gets a new status, and a new,
+# empty for_now cart is generated.
+
+# The CartChassis exists just to make things a bit tidier in the controllers
+# and views.  It doesn't (and shouldn't!) contain any unique info of its own.
+
+class CartChassis
```

Comment:
Nice, I see now. I'm not sure why we have the split, mind you -- saved for later items can expire in ways that might confuse our users (you can't hold a reservation rate by saving it for later). We'll talk tonight.

Reply:
It's mostly that processing anything other than the entire cart becomes a fragile, tangly misery-- so if we want things to be savable, This Is The Way.  It's a balancing act-- I want ways to preserve the PITA keystrokes people invest in filling out our forms, but you're right that we don't want people to be confused. 

(In this case, I'm not sure they would be-- I don't expect items I've saved for later in my Amazon cart to remain available and at the same price in perpetuity, and I think most people in 2021 have experience with that kind of thing.)

To me, it's more confusing that you can create a reservation without paying for it, and then have it possibly and silently become un-buy-able if you leave it alone for too long.